aws/s3_malware.py

```diff
@@ -0,0 +1,108 @@
+    region (us-east-1).
+
+    :param bucket_name: Bucket to create
+    :param region: String region to create bucket in, e.g., 'us-west-2'
+    :return: True if bucket created, else False
+    """
+
+    # Create bucket
+    try:
+        if region is None:
+            s3_client = boto3.client('s3')
+            s3_client.create_bucket(Bucket=bucket_name)
+        else:
+            s3_client = boto3.client('s3', region_name=region)
+            location = {'LocationConstraint': region}
+            s3_client.create_bucket(Bucket=bucket_name,
+                                    CreateBucketConfiguration=location)
+    except ClientError as e:
+        logging.error(e)
+        return False
+    return True
+
+
+def move_file_to_remediation_bucket(src_bucket_name, dest_bucket_name, old_key_name):
+    """ This function moves the malware files from the buckets to the remediation bucket
+    """
+
+    client = boto3.client('s3')
+    s3 = boto3.resource('s3')
+    copy_source = {
+        'Bucket': src_bucket_name,
+        'Key': old_key_name
+    }
+    bucket = s3.Bucket(dest_bucket_name)
+    obj = bucket.Object(old_key_name)
+    obj.copy(copy_source)
+
+    if(not version):
+        response = client.delete_object(
+            Bucket=src_bucket_name,
+            Key=old_key_name
+        )
+    else:
+        response = client.delete_object(
+            Bucket=src_bucket_name,
+            Key=old_key_name,
+            VersionId=version
+        )
+
+
+def lambda_handler(event, context):
+    """ Starting function to s3 malware remediation for malware files present in the s3 buckets.
+        Check if this function can remediate the event.
+        If no, returns
+        If yes, Move every malware file to the common remediation s3 bucket
+
+        Args:
+            event (dict):
+                This is the payload with is sent via Optix, whenever an alert is generated.
+            context (dict):
+                This is the AWS lambda context.
+            ec2_regions (list):
+                List of all available regions
+
+        Returns:
+            str: Always returns "Remediation successful". Everything else is logged.
+    """
+    create_bucket(bucket_name)
+    if event['eventType'] == 'ALERT' and event['payloadData']['alertType'] == 'S3 Malware':
+        payload_data = event['payloadData']
+        affected_resources = payload_data['affectedResources']
+        for affected_resource in affected_resources:
+            if affected_resource['state'] == "OPEN":
+                try:
+                    bucket = affected_resource['resourceInfo']
+                    affectedResourceJSON = affected_resource['affectedResourceJSON']
+                    affectedResourceJSONObject = eval(affectedResourceJSON)
+                    affectedFiles = affectedResourceJSONObject['affectedFiles']
+                    for file in affectedFiles:
+                        version=""
```

Comment:
This is why the version is failing this should be null/ None by default.

Reply:
Added.